build: Fix C4005 "macro redefinition" MSVC warnings in examples

hebasto opened the pull request:
This PR:
- fixes C4005 "macro redefinition" MSVC warnings in examples
- removes warning suppressions in both build systems, Autotools-based and CMake-based ones

real-or-random commented on examples/examples_util.h:
```diff
@@ -17,7 +17,9 @@
  */
 
 #if defined(_WIN32)
+#define WIN32_NO_STATUS
 #include <windows.h>
+#undef WIN32_NO_STATUS
 #include <ntstatus.h>
```
This is so obscure. Perhabs add a comment here.

I found https://www.mschaef.com/windows_h_is_wierd

hebasto replied:
Mind suggesting a good wording for a comment?

hebasto commented:
Updated 263fc296632060b8d2320f4382f06430fee3421e -> dc0657c7622f5a13afc3876eca7e2fc7cabb9a10 ([pr1275.01](https://github.com/hebasto/secp256k1/commits/pr1275.01) -> [pr1275.02](https://github.com/hebasto/secp256k1/commits/pr1275.02), [diff](https://github.com/hebasto/secp256k1/compare/pr1275.01..pr1275.02)):

- addressed @real-or-random's [comment](https://github.com/bitcoin-core/secp256k1/pull/1275#discussion_r1164247955)

real-or-random reviewed (approved):
utACK dc0657c7622f5a13afc3876eca7e2fc7cabb9a10

real-or-random reviewed (commented):
ACK dc0657c7622f5a13afc3876eca7e2fc7cabb9a10 I tested that the examples compile without warnings on MSVC 19.34.31942
